### PR TITLE
[PFX-558]: Remove @gasket/plugin-lifecycle

### PR DESCRIPTION
**Pull request by bbetts-godaddy**
## Summary

Ticket: [PFX-558](https://godaddy-corp.atlassian.net/browse/PFX-558)

We are moving away from magic directories in future gasket applications. As a result the `./lifecycles` directory is no longer supported, making the `@gasket/plugin-lifecycle`  plugin obsolete. 

## Changelog

- Remove `@gasket/plugin-lifecycle`

## Test Plan
Screenshot 2024-04-30 at 2 49 16 PM

- `npm run test` is passing

### CODE REVIEW

**Review by jpina1-godaddy — APPROVED** [review]
Minor nit, there are a couple readme files in the branch that still link to `/packages/gasket-plugin-lifecycle` that might need to be removed.

**Comment by agerard-godaddy** [comment]
> Minor nit, there are a couple readme files in the branch that still link to `/packages/gasket-plugin-lifecycle` that might need to be removed.

Yes noticed this, too. The PR is looking good but let's scalpel out those references and replace them with alternate guidance if needed.

**Review by mmason2-godaddy — APPROVED** [review]
Fix conflicts with upstream 👍

File: packages/gasket-plugin-webpack/docs/webpack.md [hunk 1]
```diff
@@ -20,10 +20,10 @@ brittle, so handle with care. Also, the `webpack` instance is passed via context
 with the lifecycle, providing access to the [compatible Webpack plugins]
 of the version installed.
 
-### Via lifecycle files
+### Via custom app-level plugin
 
 ```js
-// lifecycles/webpack-config.js
+// my-custom-plugin/lifecycles/webpack-config.js
```
Comment (agerard-godaddy) [inline packages/gasket-plugin-webpack/docs/webpack.md:26]: Actually, this syntax is invalid now. I think the `Via plugin hooks` example covers the required pattern if you want to drop this `Via custom app-level plugin` block

**Review by agerard-godaddy — APPROVED** [review]

**Pull request merged**